### PR TITLE
Jobsub q format fix

### PR DESCRIPTION
**Pull request by marcmengel**
So after discussing it at the meeting yesterday, I thought "How hard could it be to make the output more like old jobsub_q?"
so here it goes...
The main code involved is the logic in turning it on and off, and a few bits of classadd expression here and there.

### CODE REVIEW

File: bin/jobsub_cmd [hunk 1]
```diff
@@ -85,6 +87,33 @@ def main() -> None:
...
+        ]:
+            default_formatting = False
+
+        if (
+            i
+            in [
+                "-constraint",
+                "-factory",
+                "-autocluster",
+                "-constraint",
```
Comment (shreyb) [inline bin/jobsub_cmd:109]: I believe `-constraint` should not be in here twice.

File: bin/jobsub_cmd [hunk 2]
```diff
@@ -85,6 +87,33 @@ def main() -> None:
...
+                "-unmatchable",
+                "-allusers",
+            ]
+            or i[0] in "0123456789"
```
Comment (shreyb) [inline bin/jobsub_cmd:113]: Is this for the formatting directives like `%0.1s` or whatever those are?

File: bin/jobsub_cmd [hunk 3]
```diff
@@ -103,8 +132,62 @@ def main() -> None:
...
 
+        if default_constraint:
+            execargs.extend(
+                [
+                    "-constraint",
+                    f'''Jobsub_Group=="{os.environ['GROUP']}"''',
+                ]
+            )
+
+        if default_formatting:
```
Comment (shreyb) [inline bin/jobsub_cmd:146]: Do we want to actually default to the old jobsub format?  Or keep  the current default format, with the ability to get the old format?  That was my understanding from yesterday's meeting (I'm compiling those minutes now...sorry for the delay).  @retzkek  - how did you interpret this decision?

File: bin/jobsub_fetchlog [hunk 4]
```diff
@@ -65,6 +65,7 @@ def main():
         help='format for downloaded archive: "tar" (default, compressed with gzip) or "zip"',
         default="tar",
     )
+    parser.add_argument("job_id", nargs="?", help="job/submission ID")
```
Comment (shreyb) [inline bin/jobsub_fetchlog:68]: This should be `--jobid`, like https://github.com/marcmengel/jobsub_lite/blob/master/lib/get_parser.py#L89 specifies.  Could this be the same as the line I linked above, just to standardize the arguments across the tools?

File: templates/simple/simple.cmd [hunk 5]
```diff
@@ -36,6 +36,9 @@ when_to_transfer_output = ON_EXIT_OR_EVICT
 +JOB_EXPECTED_MAX_LIFETIME = {{expected_lifetime}}
 notify_user = {{email_to}}
 
+# set command to user executable for condor_q, etc.
++Cmd = {{executable|basename}}
+
```
Comment (shreyb) [inline templates/simple/simple.cmd:41]: Ooh - I like this.

**Review by shreyb — CHANGES_REQUESTED** [review]
Very nice - but I have a couple of change requests and questions.

File: bin/jobsub_cmd (hunk 2 above)
Comment (marcmengel) [reply bin/jobsub_cmd:113]: no, that's for ClusterId's -- if they give us a 1234.0 or 1234.0@schedd that's a constraint for a single job, so we don't want to give the (jobsub_q default) group constraint.

Comment (shreyb) [reply bin/jobsub_cmd:113]: I see.  That makes sense.  Thanks for clearing that up for me.

File: bin/jobsub_cmd (hunk 3 above)
Comment (marcmengel) [reply bin/jobsub_cmd:146]: Yes, I was going to bring this up for discussion again; Should we not default to the old behavior? If people are reading current jobsub output, say in a script, then we should keep the old output... do we think users are wrapping jobsub_q or parsing its output in other scripts?

Comment (shreyb) [reply bin/jobsub_cmd:148]: Resolved - jobsub working group decided to have default behavior match old jobsub.  So this is good.

File: bin/jobsub_fetchlog (hunk 4 above)
Comment (marcmengel) [reply bin/jobsub_fetchlog:68]: So... we take *either*` --jobid 1234@schedd `*or* just `1234@schedd`   the former is "jobid", the latter is "job_id", and after we parse the arguments we set one from the other if it is not set.

File: bin/jobsub_cmd (hunk 1 above)
Comment (marcmengel) [reply bin/jobsub_cmd:108]: fixed.

File: bin/jobsub_fetchlog (hunk 4 above)
Comment (shreyb) [reply bin/jobsub_fetchlog:68]: I did miss that - thanks for pointing it out.  In that case, I'm good with this.

**Review by shreyb — APPROVED** [review]
Looks good - thanks!

**Pull request merged**